ARCs/arc-draft_action_routing.md

```diff
@@ -0,0 +1,66 @@
+---
+arc: <to be assigned>
+title: Application Action Routing
+description: Standardized handling of OnCompletion and ApplicationID actions 
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2024-01-02
+---
+
+## Abstract
+This ARC provides a standard way to handle an application call depending on the combination of the `OnCompletion` and `ApplicationID`.
+
+## Motivation
+Currently there is no way for someone to tell whether an app can be updated or deleted without doing a full static analysis on the TEAL.
+
+## Specification
+
+### TEAL
+The TEAL for an application **MUST** start with the following lines:
+
+```
+txn ApplicationID
+!
+int 6
```

Comment:
When you disassemble a contract, you won't see `int 6`.  You might see `intc X` or `pushint 6`

Reply:
Ah right. My understanding is that an explicit `pushint` doesn't get added to the constant block, is that true? If so I can change the TEAL to use `pushint`